src/Dashboard/Grid.php

```diff
@@ -1138,6 +1142,56 @@ public function getAllDasboardCards($force = false): array
                 }
             }
 
+            foreach ($CFG_GLPI['itemdevices'] as $itemtype) {
+                if (property_exists($itemtype, 'itemtype_2')) {
```

Comment:
Instead of using `itemtype_2` property, you should rely on `$itemtype::getDeviceForeignKey()`.

Also, I guess an `Item_Devices` implementation has always a related device class.

Reply:
indeed it works without the check